### PR TITLE
Tech isolv bse

### PR DESCRIPTION
**Pull request by sangallidavide**
* New tddft implementation for eh_space. Collisions and kernel are computed in G-space. Older implementation in R-space is kept for checking purpposes.
```
- TDDFT_ALDA_R_space.o
+ TDDFT_ALDA_eh_space_G_collisions_L.o
+ TDDFT_ALDA_eh_space_G_collisions_R.o
+ TDDFT_ALDA_eh_space_G_kernel.o
+ TDDFT_ALDA_eh_space_R_kernel.o
```
The new implementation is much faster for 2 reasons:
a) Collisions are computed before and then assembled together, thus 2N times instead of N^2 times
b) Collisions and kernel are done in G-space, and ng_closed << fft_size. Moreover a cutoff can be used

* New slepc algorithm takes into account the structure of the BSE matrix with coupling. Developed by the slepc team. See #66 

*  Configure upgraded to work with development version of slepc, and ndb.BS_PAR file now contains also the BS_energies and the BS_table. This makes it possible to fully reconstruct the BSE externally to yambo.  (See #83 )

### CODE REVIEW

File: src/interface/INIT_activate.F
Comment (andreamarini) [inline src/interface/INIT_activate.F:1]: I know it was like this even before :) But is it possible to use more human-understandable variable name? def/full/cut_Gmax/cut_GmGp seems a little bit cryptic to me

Comment (sangallidavide) [reply src/interface/INIT_activate.F:1]: Let's put this on the todo list for future developments.

**Comment by sangallidavide** [comment]
The robot are green in serial, there is one test failing in parallel. I'm checking this

https://media.yambo-code.eu/robots/tech-isolv-bse/unimi-XPS-8930.php

**Comment by sangallidavide** [comment]
> The robot are green in serial, there is one test failing in parallel. I'm checking this
> 

Fixed bug in parallel. Now the robots are green:

[https://media.yambo-code.eu/robots/tech-isolv-bse/unimi-XPS-8930.php](https://media.yambo-code.eu/robots/tech-isolv-bse/unimi-XPS-8930.php)

**Review by attacc — APPROVED** [review]
Fine for me

**Comment by attacc** [comment]
Just to understand these changes require a special branch of PETSc/Slepc?
And I guess the flag petsc-branch will disappear when a new version of PETSc will be available?

**Comment by sangallidavide** [comment]
> Just to understand these changes require a special branch of PETSc/Slepc? 

All the changes are already available with the released Petsc/Slepc version linked in yambo

**Comment by attacc** [comment]
Ciao Davide

may you test these changes with INTEL I got a fail in the test: 

C6H3Cl3

[serial] 06_BSE_cpl_slepc1:  optics T-space static screening     4.5s
  o-06_BSE_cpl_slepc1.alpha_q1_slepc_bse          [  NO in OUTPUT ]
  o-06_BSE_cpl_slepc1.jdos_q1_slepc_bse           [  NO in OUTPUT ]

ciao
Claudio

**Comment by sangallidavide** [comment]
Which version of the slepc library did you use?

**Comment by attacc** [comment]
You are right... I'm using an old version... I will update it and run it again

**Comment by sangallidavide** [comment]
I'll see if I can add a check on the slepc version

**Pull request merged**